Add config for community workflow test on AWS

christopherwharrop-noaa opened the pull request:
## DESCRIPTION OF CHANGES: 
Adds the configuration script for generating community workflow on AWS.  The configuration uses fix files in a permanent location (`/contrib/NCEPDEV/global/glopara/fix`) on Parallel Works, but users are required to copy the initial conditions from the S3 bucket as described in the [instructions](https://github.com/ufs-community/ufs-srweather-app/wiki/Getting-Started).

This PR also:

1.  Adds automatic calculation of `OMP_NUM_TRHEADS` to correct value
2. Adds automatic calculation of `--cpus-per-task` to the correct value.
3. Enables users to override default IO quilting settings
4. Updates 3km config for us in AWS cloud RRFS experiments with correct run commands, layout, and quilting.

This PR is dependent on NOAA-GSL/ufs-srweather-app#14 and should be merged at the same time.

## TESTS CONDUCTED: 
Ran end-to-end community workflow on the AWS Parallel Works platform using Rocoto.  Verified all tasks completed successfully.

## CONTRIBUTORS (optional): 
The `config.community.aws.sh` was adapted from work done by @christinaholtNOAA

christopherwharrop-noaa committed:
Add config for community workflow test on AWS

JeffBeck-NOAA reviewed (approved):
Just a note that in the authoritative repo we don't store platform-specific config files within the repository (only the two community and NCO mode scripts).  But it's fine here.

christinaholtNOAA reviewed (approved):
LGTM! Thanks for adding this config for our reference.

christopherwharrop-noaa committed:
Enable correct workflow generation for AWS

1. Calculate correct value for OMP_NUM_THREADS automatically.
2. Calculate correct value for --cpus-per-task for forecast automatically.
3. Enable users to override defaults for quilting options.
4. Update community config for AWS with correct layout/quilting and run commands.
5. Update 3km conus config for AWS with correct layout/quilting and run commands.

christopherwharrop-noaa commented:
@christinaholtNOAA - You may want to take another look here.  I added another commit that makes several important changes that allows for workflows to be correctly generated such that no manual edits are needed.

christinaholtNOAA commented on ush/config.sh.aws_cloud:
```diff
@@ -1,79 +1,84 @@
+EXPT_SUBDIR="test_rrfs"
+
+RUN_CMD_UTILS="srun --mpi=pmi2"
+RUN_CMD_FCST="srun --mpi=pmi2"
```
I was thinking this setting landed on mpirun instead.

christopherwharrop-noaa replied:
There was some initial confusion about this.  But, I think the problem was that I wasn't using the correct Intel MPI module.   And that has been corrected in NOAA-GSL/ufs-srweather-app#16.  Now, I am getting much better timings and no hangs.

christinaholtNOAA reviewed (approved):
@christopherwharrop-noaa These changes look great. Thanks for doing that. Just one question below.

christopherwharrop-noaa commented:
I see that  NOAA-GSL/ufs-srweather-app#16 was merged.  Can this one be merged now, too?
